sync.py

```diff
@@ -253,19 +258,37 @@ def write_jenkins_file():
     # name is mandatory, so we use the filename if nothing is set in a template
     if args.verbose:
         print(ET.tostring(template))
+    if template.find('category') is not None and template.find('category').text not in categories:
+        c = template.find('category').text
+        template.remove(template.find('category'))
+        if args.verbose:
+            print(f'WARNING: Unable to find category "{c}" in the JSS, setting to None')
     if template.find('name') is None:
         ET.SubElement(template, 'name').text = script
     elif template.find('name').text is '' or template.find('name').text is None:
         template.find('name').text = script
     return template
 
 
+async def get_existing_categories(session, url, user, passwd, semaphore):
+    auth = aiohttp.BasicAuth(user, passwd)
+    headers = {'content-type': 'application/xml'}
+    async with semaphore:
+        with async_timeout.timeout(10):
```

Comment:
Do you want to set this to args.timeout? 
```suggestion
        with async_timeout.timeout(args.timeout):
```